http://patchwork.kernel.org/patch/55516/

1. This patch is based on the kernel of 2.6.32-rc4 
2. In this patch, we enlarge the out buffer size to optimize the upload speed for the ppp connection. Then it can support the upload of HSUPA data cards.

Signed-off-by: fangxiaozhi <huananhu@huawei.com>

3. Also, we have to modify serial_post_job() for 2.4 kernel

--- a/drivers/net/ppp_async.c	2009-10-12 05:43:56.000000000 +0800
+++ b/drivers/net/ppp_async.c	2009-10-15 16:29:56.000000000 +0800
@@ -36,7 +36,7 @@
 
 #define PPP_VERSION	"2.4.2"
 
-#define OBUFSIZE	256
+#define OBUFSIZE	2048
 
 /* Structure for storing local state. */
 struct asyncppp {
--- a/drivers/usb/serial/usbserial.c	2008-12-04 19:16:55.000000000 +0300
+++ b/drivers/usb/serial/usbserial.c	2009-11-07 11:03:25.000000000 +0300
@@ -810,7 +810,7 @@ static int serial_post_job(struct usb_se
 	if (port == NULL)
 		return -EPIPE;
 
-	if (count >= 512) {
+	if (count > 2048) {
 		static int rate = 0;
 		/*
 		 * Data loss due to extreme circumstances.
@@ -822,7 +822,7 @@ static int serial_post_job(struct usb_se
 			err("too much data (%d) from %s", count,
 			    from_user? "user": "kernel");
 		}
-		count = 512;
+		count = 2048;
 	}
 
 	while (done < count) {
